lang/funcs/collection.go

```diff
@@ -634,6 +635,144 @@ var MatchkeysFunc = function.New(&function.Spec{
 	},
 })
 
+// MergeFunc contructs a function that takes an arbitrary number of maps and
+// returns a single map that contains a merged set of elements from all of the maps.
+//
+// If more than one given map defines the same key then the one that is later in
+// the argument sequence takes precedence.
+var MergeFunc = function.New(&function.Spec{
+	Params: []function.Parameter{},
+	VarParam: &function.Parameter{
+		Name:             "maps",
+		Type:             cty.DynamicPseudoType,
+		AllowUnknown:     true,
```

Comment:
I don't think `AllowUnknown` is necessary here, since if any of the input values are unknown then the result would necessarily be unknown itself, right? (The `ElementIterator` calls inline would panic, then.)